plugins/hls-hlint-plugin/src/Ide/Plugin/Hlint.hs

```diff
@@ -167,9 +167,8 @@ rules :: PluginId -> Rules ()
 rules plugin = do
   define $ \GetHlintDiagnostics file -> do
     config <- getClientConfigAction def
-    let pluginConfig = configForPlugin config plugin
-    let hlintOn' = hlintOn config && plcGlobalOn pluginConfig && plcDiagnosticsOn pluginConfig
-    ideas <- if hlintOn' then getIdeas file else return (Right [])
+    let hlintOn = pluginEnabledConfig plcDiagnosticsOn plugin config
+    ideas <- if hlintOn then getIdeas file else return (Right [])
```

Comment:
I have no idea how this works :sweat_smile: If I do that then the test that changing the config works starts timing out :shrug:

Reply:
hmm ok i think we should change `noHlintDiagnostics` for

```haskell
noHlintDiagnostics :: Assertion
noHlintDiagnostics  = expectNoMoreDiagnostics 3 doc "hlint"
```

to avoid the parser hang on some diagnostic

Reply:
Do you mind if I leave that as a future improvement, and leave this PR as just getting rid of the old settings?